File: modules/bar/sysinfo.ts
import { Widget, Roundedges } from "imports"
import { VolumeIndicator, Battery } from "../Widgets/index"
import NetworkButton from "./Network"
import BluetoothButton from "./Bluetooth"

const { RoundedAngleEnd } = Roundedges;
const { Box, Button } = Widget;

const SysInfoBox = () => Box({
	hexpand: true,
	class_name: 'sysinfo',
	spacing: 8,
	children: [
		VolumeIndicator(),
		BluetoothButton(),
		NetworkButton(),
		Battery(),
	]
});

export const SysInfo = () => Box({
	children: [
		RoundedAngleEnd("topleft", { class_name: "angleLeft" }),
		SysInfoBox(),
		RoundedAngleEnd("topright", { class_name: "angleRight" }),
	]
})

File: Rust/Modules/Bar/SysInfo.rs
r
use imports::{Widget, Roundedges};
use Widgets::index::{VolumeIndicator, Battery};

let RoundedAngleEnd = Roundedges::RoundedAngleEnd;
let Box = Widget::Box;

fn sys_info_box() -> Box {
    Box::new()
        .hexpand(true)
        .class_name("sysinfo")
        .spacing(8)
        .children(vec![
            VolumeIndicator::new(),
            Battery::new(),
        ])
}

pub fn sys_info() -> Box {
    Box::new()
        .children(vec![
            RoundedAngleEnd("topleft", vec![("class_name", "angleLeft")]),
            sys_info_box(),
            RoundedAngleEnd("topright", vec![("class_name", "angleRight")]),
        ])
}

